tests/system/action/user/test_email_client_server.py

```diff
@@ -0,0 +1,510 @@
+            ssl_context = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+            ssl_context.load_default_certs(purpose=ssl.Purpose.SERVER_AUTH)
+            ssl_context.load_cert_chain("cert.pem", "key.pem")
+            ssl_context.check_hostname = False
+            ssl_context.verify_mode = ssl.VerifyMode.CERT_NONE
+            if EmailSettings.connection_security == ConnectionSecurity.SSLTLS:
+                # This is a hack: The aiosmtpd library does not issue AUTH in EHLO, if not starttls is used.
+                # For other methods (SSL/TLS and NONE) setting auth_require_tls allows AUTH. The intention is to
+                # allow AUTH before TLS (which is ok for NONE), but a hack for SSL/TLS since we have an
+                # encrypted connection
+                if self.auth:
+                    auth_kwargs["auth_require_tls"] = False
+                self.controller = Controller(
+                    self.handler,
+                    EmailSettings.host,
+                    EmailSettings.port,
+                    server_hostname="127.0.0.1",
+                    ssl_context=ssl_context,
+                    **auth_kwargs,  # type: ignore
+                )
+            else:
+                self.controller = Controller(
+                    self.handler,
+                    EmailSettings.host,
+                    EmailSettings.port,
+                    server_hostname="127.0.0.1",
+                    require_starttls=True,
+                    tls_context=ssl_context,
+                    **auth_kwargs,  # type: ignore
+                )
+        else:
+            if self.auth:
+                auth_kwargs["auth_require_tls"] = False
+            self.controller = Controller(
+                self.handler, EmailSettings.host, EmailSettings.port, **auth_kwargs  # type: ignore
+            )
+
+        self.controller.start()
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_value: Optional[BaseException],
+        exc_traceback: Optional[TracebackType],
+    ) -> None:
+        self.controller.stop(no_assert=True)
+
+
+class Authenticator:
+    def __init__(self) -> None:
+        pass
+
+    def __call__(
+        self,
+        server: SMTP,
+        session: Session,
+        envelope: Envelope,
+        mechanism: str,
+        auth_data: LoginPassword,
+    ) -> AuthResult:
+        fail_nothandled = AuthResult(success=False, handled=False)
+        if mechanism not in ("LOGIN", "PLAIN"):
+            return fail_nothandled
+        if not isinstance(auth_data, LoginPassword):
+            return fail_nothandled
+
+        if auth_data.login == b"sender@example.com" and auth_data.password == b"secret":
+            return AuthResult(success=True)
+        else:
+            return fail_nothandled
+
+
+class SendMailWithSmtpServer(BaseActionTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.create_meeting()
+        self.set_models(
```

Comment:
The users are not needed since you are only testing the email mixin and never the action

Reply:
Done